src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -94,6 +94,7 @@
                     + "                                         ERROR,WARN,INFO,DEBUG,TRACE [default: INFO].\n"
                     + " -Q, --quickcheck-test-mode              Run in QuickCheck test mode\n"
                     + " -P <seconds>, --cm-poll-interval        Configuration manager poll interval [default: 1]\n"
+                    + " -v, --verify                            Compute and verify checksums for log entries.\n"
```

Comment:
@slfritchie you make a good point about lazily dealing with data corruption and rejecting access to all data if an error is found. 

Also, startup doesn't necessarily mean scanning in the constructor (that's just code organization), you can start the server that doesn't read/write, but responds to other requests. Also, the scan can use bounded resources, so you don't have to allocate resources proportional to the log size.